### ISSUE TITLE
Time zone mishandled by sync

### ISSUE THREAD

**Issue by patricktokeeffe** [owncloud/calendar#797]
Using v0.6.4 on ownCloud 8.0.3 with Android 5.0.2, Davdroid 0.7.6, and AOSP calendar standalone-1.2:

Phone uses network time zone (PDT) and calendar is set to Los Angeles (PDT). When I create an event the timezone is initially shown (PDT=GMT-7) then it's hidden after I enable the 'all day' option. I save the new all day event and it's properly shown as all day on my calendar.

After refreshing (syncing) the all day event is shifted back 7 hours: 1700 of the previous day to 1700 of target day. This shift only applies to all day events created on my phone, not events with explicit times or any event created through the web interface. 

Does owncloud calendar think the android all day events are UTC-based? Is there a way I could check what it's actually receiving?

**Comment by patricktokeeffe** [comment]
I'm guessing it's AOSP calendar not sending a timezone with the event. ?? Changing the phone's timezone affects all the 'shifted' all-day events. And events created in the owncloud calendar web interface or TB Lightning render correctly.

How about this RFE?: Assign the calendar's timezone to events which lack one

**Comment by cuthulino** [comment]
I got nearly the same problem,
i am using aCalendar and when i change an event to "all day" it removes the timezone field to change it.
when i syn the calendar, owncloud gets an two day event from 2:00 to 2:00

that is realy enerving...

**Comment by georgehrke** [comment]
Please provide an example ics 

Please excuse my brevity and typos.

**Comment by cuthulino** [comment]
Here is the ics of an all day event made with the app "acalendar":
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:ownCloud Calendar 0.6.4
X-WR-CALNAME:Persönlich-Stefan
BEGIN:VEVENT
DTSTAMP:20150619T083024Z
UID:20150619T083024Z-21909@77d46c2e7662503c
DTSTART;TZID=Europe/London:20150621T010000
DTEND;TZID=Europe/London:20150622T010000
SUMMARY:Testevent
STATUS:CONFIRMED
LAST-MODIFIED:20150619T083024Z
END:VEVENT
END:VCALENDAR
```

**Comment by georgehrke** [comment]
The first one is simply no correct allday-event, because it contains time information, whereas the second does not. this is not a bug in ownCloud, but a bug in acalendar.

**Comment by cuthulino** [comment]
i dont think so, acalender creates a one day event, only when the event is sent to the owncloud it gets an two day event with times.
before all is correct, so owncloud maybe dont understand the sent event!?

**Comment by georgehrke** [comment]
> before all is correct, so owncloud maybe dont understand the sent event!?

It would display it correctly, if the timezone was correct. But it's London, which is, as you said, incorrect. ownCloud doesn't modify the ics when writing it to the db, so this is definitely a bug in acalendar.
Even if the timezone would be correct, ownCloud would show an event that starts at 12am and ends 12am the next day, and not a proper all-day event, because the data contains a time. If it would be a proper all-day event, it would contain no time.

**Comment by cuthulino** [comment]
ok thanks
i will reply to their devs.
